### PR TITLE
User Navigation and Gas Station Refactor

### LINKED ISSUE #2274: Wallet Controls / Gas Station Visual Refresh
_Note: This issue relies on #2267 to provide the grid layout_

## Specification

Discernible features:
- As opposed to the current iteration, it will feature just the user avatar and dropdown menu, notifications icon, and, depending on the state of the wallet (connected or not), a button to connect the wallet, or the user's wallet address, in shorthand form
- When the wallet is not connected, a "Connect Wallet" button is displayed _(see the screenshots below)_
- When the wallet is not connected, the currently connected wallet's address is displayed _(see the screenshots below)_, in shorthand form
- If there's a transaction to be signed, a ~red~ pink number display dot is displayed, the number representing the number of transactions needed to be signed _(individual transactions or transaction groups -- this functions as it is now)_
- The user dropdown menu remains the same as it is now
- The gas station remains the same as it is now
- The connect wallet popup stays the same as now but wired into the "Connect Wallet" button
- Please observe the various hover state colors

Since this PR does not require any additional logic to be added _(as everything is in there)_, it should be visually and functionally complete when this issue is implemented

### Design

See Figma section: Home UI / Actions, Default View, Connect wallet button, Wallet popover - active state, _and_ Home UI/Events Log, both screens: [Colony Simplified Figma Screens](https://www.figma.com/file/qQoBbyyDXkuHwoqHIOGjEu/Colony-Simplified)

Details view

Wallet not connected

Screenshot from 2020-10-05 17-01-28

Screenshot from 2020-10-05 17-01-09

Transaction(s) need signing

Screenshot from 2020-10-05 17-01-55

Hover state

Screenshot from 2020-10-05 17-02-07

Connect a wallet popover (the currently existent one)

Screenshot from 2020-10-05 17-01-37

Sign a transaction (the current gas station)

Screenshot from 2020-10-05 17-01-19

Context view

Screenshot from 2020-10-05 16-46-01

### PR DESCRIPTION
**Pull request by rdig**
## Description

This PR refreshes the user navigation controls, adds new features, and refactors a whole lot of logic on how the the Gas Station controls used to work

**Changes**

- [x] Removed all hardware wallets
- [x] Removed all software wallets
- [x] Removed ability to create a new wallet
- [x] Removed user navigation route/icon to dashboard
- [x] Redesigned wallet connect button
- [x] Redesigned user wallet connected button _(now shows the user's address)_
- [x] Redesigned inbox notification icon
- [x] Refactored all popovers placements
- [x] Gas station now automatically signs transactions
- [x] Gas station now automatically shows the details of the current transaction that needs to be signed
- [x] Added Gas station notification icon
- [x] Removed gas station gas pricing controls and _"Confim"_ button
- [x] Added a bunch of helper methods
- [x] Added network info display
- [x] Added warning if wallet provider connected to the wrong network
- [x] A bunch of safe-guards against users creating transactions when connected to the wrong provider

**Reviewer Note**

Sorry for the length of this, it went through so many spec changes... :(

**Demo**

cs-user-navigation

Resolves DEV-4
Resolves #2274

### CODE REVIEW

**Review by ceolson01 — APPROVED** [review]
Whew! Lots of cleanup - well done 👏 

No major issues, just an enhancement suggestion or two and a couple style bugs that I trust you can sort out yourself prior to merging 🙂

File: src/modules/pages/components/RouteLayouts/UserNavigation.tsx [hunk 1]
```diff
@@ -1,98 +1,138 @@
...
-      <AvatarDropdown />
+              <div
+                className={`${styles.notificationsIcon} ${
+                  isOpen ? styles.notificationsIconActive : ''
+                }`}
+              >
+                <InboxIcon
+                  notifications={notifications}
+                  title={MSG.inboxTitle}
+                />
```
[the remark this answers is not in the document]
Comment (rdig) [reply src/modules/pages/components/RouteLayouts/UserNavigation.tsx:129]: I don't see this... hmmm

[the remark this answers is not in the document]
Comment (rdig) [reply src/modules/pages/components/RouteLayouts/UserNavigation.tsx:129]: > for keyboard users

If you can convince Karol and Jack about this, I'll put it in, until then it stays as it is _(as I've mentioned in the other PR, I've lost the argument for this)_

File: src/modules/pages/components/RouteLayouts/UserNavigation.tsx [hunk 2]
```diff
@@ -1,98 +1,138 @@
...
-            type="button"
-            className={styles.navigationItemButton}
-            ref={ref}
-            onClick={toggle}
-          >
-            <div
-              className={`${styles.navigationItem} ${
-                isOpen ? styles.navigationItemActive : ''
-              }`}
+              <MaskedAddress address={walletAddress} />
```
[the remark this answers is not in the document]
Comment (rdig) [reply src/modules/pages/components/RouteLayouts/UserNavigation.tsx:102]: Sure

File: src/modules/pages/components/RouteLayouts/UserNavigation.tsx [hunk 3]
```diff
@@ -1,98 +1,138 @@
...
-              className={`${styles.navigationItem} ${
-                isOpen ? styles.navigationItemActive : ''
-              }`}
+      {userCanNavigate && (
+        <div
+          className={styles.networkInfo}
+          title={isNetworkAllowed && ALLOWED_NETWORKS[networkId || 1].name}
+        >
+          {isNetworkAllowed && ALLOWED_NETWORKS[networkId || 1].shortName}
+        </div>
```
[the remark this answers is not in the document]
Comment (rdig) [reply src/modules/pages/components/RouteLayouts/UserNavigation.tsx:64]: Nope. Not in spec. Title is the best I could do, to show some semblance of a full network name

File: src/modules/pages/components/RouteLayouts/UserNavigation.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (ceolson01) [reply src/modules/pages/components/RouteLayouts/UserNavigation.tsx:129]: > If you can convince Karol and Jack about this, I'll put it in, until then it stays as it is (as I've mentioned in the other PR, I've lost the argument for this)

Challenge accepted 😎 when the new week rolls around, I'll schedule a call with them - ignore this for now (along with the other PRs where I brought this up, for now - a11y can be tackled post-merge)

[the remark this answers is not in the document]
Comment (rdig) [reply src/modules/pages/components/RouteLayouts/UserNavigation.tsx:129]: Turns out you were right:

Screenshot from 2020-10-31 11-25-24

Screenshot from 2020-10-31 11-27-55

It was a Chrome v. Chromium issue _(and I assume Brave as well, as I remember you running that)_.

Anyway, they are now fixed:

Chrome:
Screenshot from 2020-10-31 12-27-38

Chromium:
Screenshot from 2020-10-31 12-27-45

**Pull request merged**